mayavi/core/ui/engine_dock_pane.py

```diff
@@ -0,0 +1,33 @@
+from pyface.tasks.api import TraitsDockPane
+from traits.etsconfig.api import ETSConfig
+
+
+class EngineDockPane(TraitsDockPane):
+
+    id = 'engine_pane'
+    name = 'Mayavi Engine'
+
+    def _model_default(self):
+        from mayavi.core.ui.engine_view import EngineView
+        return EngineView(engine=self._get_engine(self.task.window))
+
+    def _get_engine(self, window):
+        """Return the Mayavi engine of the particular window."""
+        from mayavi.core.engine import Engine
+        return window.application.get_service(Engine)
```

Comment:
There is a general philosophy when building Envisage apps that going through the `application` to get something within a component is a hint that things aren't architected quite right.  At a minimum it makes testing much harder, because now you need to have an (Envisage) app that has an `Engine` service available (or mock it).  More generally it makes things harder to re-use or change, because things are more tightly linked.

The approach that works better is to have things passed down from the application when they are created.  I would re-engineer this class to have an `engine` trait (or maybe even make the `engine` the `model`) and then when you create the dockpane here: https://github.com/enthought/mayavi/blob/622dc0542dc5c090ab0b08c1f3c670c8d383c88e/mayavi/plugins/mayavi_ui_tasks_plugin.py#L203
instead do the get service there (or even better, have the engine in turn be state of the `Task`, and pass the engine in to the task factory from the plugin).